### PR TITLE
gages added to map with color

### LINKED ISSUE #10: Add site locations to the map
and make sure they are colored appropriately, if and when color process step is complete (#19)

### PR DESCRIPTION
**Pull request by wdwatkins**
`test_plot_funs(watermark_fun, basemap_fun, datetime_fun_20180912_00, gage_sites_plot_fun_20180912_00, png_file = 'test.png')`
image

fixes #10

### CODE REVIEW

File: 6_visualize/src/prep_gage_sites_fun.R [hunk 1]
```diff
@@ -1,25 +1,13 @@
...
-    plot(sf::st_geometry(missing_data_sites$geometry),
-         pch = 4, add = TRUE,
-         col = gage_col_config$gage_out_col, lwd = 4, cex = 1.3)
-
+prep_gage_sites_fun <- function(percentile_color_data_ind, sites_sp, gage_style, dateTime){
+  this_date <- as.POSIXct(dateTime, tz = "UTC") #watch timezones if we ever switch from daily data
+  #only plotting sites with color for now
+  percentile_color_data <- readRDS(as_data_file(percentile_color_data_ind))
+  this_date_colors <- filter(percentile_color_data, dateTime == this_date)
+  sites_sp@data <- left_join(sites_sp@data, this_date_colors, by = "site_no")
```
[the remark this answers is not in the document]
Comment (lindsaycarr) [reply 6_visualize/src/prep_gage_sites_fun.R]: I think it should probably happen in the processing step to get colors added. Right now, we are filtering out NAs.

File: 6_visualize/src/prep_gage_sites_fun.R [hunk 2]
```diff
@@ -1,25 +1,13 @@
...
+  gage_style <- gage_style$gage_style
```
Comment (lindsaycarr) [inline 6_visualize/src/prep_gage_sites_fun.R]: I'm not really following how `gage_style$gage_style` is needed. It looks like you pass in the result of `viz_config[[I('gage_style')]]` to this function, so you should be able to just say `gage_style$pch`, etc. Am I missing something?

Comment (wdwatkins) [reply 6_visualize/src/prep_gage_sites_fun.R]: I thought that too, but if you put a `browser()` here, its ending up with an extra list level of `gage_style`.  I'm not sure why.

Comment (lindsaycarr) [reply 6_visualize/src/prep_gage_sites_fun.R]: Huh, ok. Not the end of the world, but I followed the same pattern and didn't have that issue

Comment (aappling-usgs) [reply 6_visualize/src/prep_gage_sites_fun.R]: Hypothesis: David, could it be the case that the first time you modified 6_visualize.yml, you had
```yml
  gage_style_config:
    command: viz_config[I('gage_style')]
```
and then when you changed it to double brackets (`command: viz_config[[I('gage_style_config')]]`), you still were getting a length-one list with an element named `gage_style_config`? If that's the case, you should force a rebuild of `gage_style` with `scmake('gage_style_config', force=TRUE)`. It's a remake bug that remake doesn't notice when you switch between single and double brackets within a remake command.

Comment (aappling-usgs) [reply 6_visualize/src/prep_gage_sites_fun.R]: Result: I deleted the one line Lindsay called out above and am now able to build a GIF successfully with the expected plot formatting. I'll make that change in my PR I'm about to submit.

Comment (wdwatkins) [reply 6_visualize/src/prep_gage_sites_fun.R]: thought there must be something fishy going on

**Pull request merged**